src/celeritas/Quantities.hh

```diff
@@ -4,116 +4,50 @@
 // SPDX-License-Identifier: (Apache-2.0 OR MIT)
 //---------------------------------------------------------------------------//
 //! \file celeritas/Quantities.hh
-//! \brief Unit struct definitions for use with the Quantity class
+//! \brief Derivative unit classes and annotated Quantity values
 //---------------------------------------------------------------------------//
 #pragma once
 
+#include "celeritas_config.h"
 #include "corecel/math/Quantity.hh"  // IWYU pragma: export
 
-#include "Constants.hh"
-#include "Units.hh"
+#include "UnitTypes.hh"
 
 namespace celeritas
 {
 namespace units
 {
 //---------------------------------------------------------------------------//
-//! Special case for annotating that values are in the native unit system
-struct NativeUnit
-{
-    //! The conversion factor of the resulting unit is always unity
-    static CELER_CONSTEXPR_FUNCTION int value() { return 1; }
-};
-
-//! Unit for quantity such that the numeric value of 1 MeV is unity
-struct Mev
-{
-    //! Conversion factor from the unit to CGS
-    static CELER_CONSTEXPR_FUNCTION real_type value()
-    {
-        return real_type(1e6) * constants::e_electron * units::volt;
-    }
-    //! Text label for output
-    static char const* label() { return "MeV"; }
-};
-
-//! Unit for annotating the logarithm of (E/MeV)
-struct LogMev
-{
-    //! Conversion factor is not multiplicative
-    static CELER_CONSTEXPR_FUNCTION real_type value() { return 1; }
-};
-
-//! Unit for relativistic speeds
-struct CLight
-{
-    //! Conversion factor from the unit to CGS
-    static CELER_CONSTEXPR_FUNCTION real_type value()
-    {
-        return constants::c_light;
-    }
-};
+//!@{
+//! \name Derivative units
 
-//! Unit for precise representation of particle charges
-struct EElectron
+//! "Natural units" for momentum
+struct MevPerC : UnitDivide<Mev, CLight>
```

Comment:
What is the advantage of having this 3 units types defined in `Quantities.hh` rather than with the other unit types (i.e. in `UnitTypes.hh`)?

Reply:
The only reason was that `UnitDivide` is defined in `Quantity.hh` which isn't included by `UnitTypes`, but maybe I should either remove those helpers or move them somewhere else...

Reply:
> move them somewhere else...

+1